### DISCUSSION TITLE
Data Object Download Issue In PXP

### DISCUSSION THREAD

**Discussion by KasthuriAnandhan** [pimcore/pimcore#16562] [Q&A / Support]
Hi All, 

Could you please anyone help me here to resolve the below issue.

Can't to download more than one data object in Pimcore's Product Experience Portal. Able to download single data object but when I try more than one data object can't to do that. The generation for download have been started but that will not get completion. For long time it's showing like below image. Is anybody have an idea about to resolve this issue please give a solution to me.
Thanks!!

![image](https://github.com/pimcore/pimcore/assets/125338676/e1cc396f-2def-4395-b348-ed655008fcd4)

**Comment by kingjia90** [comment]
There are no known problems/bugs so far on this aspect.

Please check if the workers are configured accordingly as per https://pimcore.com/docs/platform/Portal_Engine/Administration_of_Portals/Background_Processes

**Comment by KasthuriAnandhan** [comment]
Okay will try and let you know thank you so much for your timely response.

**Comment by arulselvan008** [comment]
Hi @kingjia90 , @brusch , @fashxp 

I have tried running the command **php bin/console messenger:consume portal-engine-async**.
![cmd](https://github.com/pimcore/pimcore/assets/106297450/437b1509-8883-4697-b173-60a906f9d331)

Single data object export is working, but multi data object download only gets the issue.

Once the download process is complete, the download button appears. However, when I click the download button, I receive a permission denied error.![Permission Error](https://github.com/pimcore/pimcore/assets/106297450/39ed35fb-2baf-490e-b849-ffd7244036dc)

I have already tried the following commands:
**chmod 755 /var/www/pimcore/var/tmp/portal-engine/downloads**
**chown www-data:www-data /var/www/pimcore/var/tmp/portal-engine/downloads**
Could you please check if the workers are configured correctly as per the documentation: https://pimcore.com/docs/platform/Portal_Engine/Administration_of_Portals/Background_Processes

Does anyone have an idea for this error?

Thank you for your assistance.

**Reply by kingjia90** [reply]
Hi, i'd try again the chmod and chown commands but with the recursive (-R) flag

**Reply by fashxp** [reply]
And make sure your messerger processes run as same user as your fpm processes ...

**Reply by arulselvan008** [reply]
Thanks for your immediate response. Let me check and come again.

**Reply by arulselvan008** [reply]
Thanks @kingjia90 @fashxp 

We have debugged the file permission issue. Initially, the root user was created, but we then changed the user to www-data and granted all necessary permissions to the path.

sudo chown -R www-data:www-data /var/www/pimcore/var/tmp/portal-engine/downloads
sudo chmod -R 755 /var/www/pimcore/var/tmp/portal-engine/downloads

We ran the messenger processes manually and now it’s set up in the cron job.

sudo -u www-data crontab -e
/bin/nano
*/5 * * * * /usr/bin/php /var/www/pimcore/bin/console messenger:consume portal-engine-async

**Reply by arulselvan008** [reply]
@kingjia90 @fashxp 

I have one more question regarding the download function for DataObjects. Currently, it only downloads data for DataObjects and does not include data from **ObjectBricks**.

How can we include the mapped ObjectBricks data with the DataObject download function?
